cabal.project

```diff
@@ -37,3 +37,38 @@ benchmarks: true
 
 program-options
   ghc-options: -Werror
+
+if impl(ghc >= 9.6)
+  allow-newer:
+    , *:base
+    , *:deepseq
+    , *:ghc-prim
+    , aeson:template-haskell
+    , hedgehog:template-haskell
+    , recursion-schemes:template-haskell
+    , scientific:template-haskell
+    , text-short:template-haskell
+    , uuid-types:template-haskell
+    , vector-th-unbox:template-haskell
+
+  constraints:
+    , aeson >= 2.0.0.0
+    , hastache >= 0.6.1
+    , monad-par >= 0.3.5
+    , statistics >= 0.16.2.0
+
+-- `aeson` has a whole bunch of dependencies (and sometimes the dependency's dependency) which do
+-- not compile with `ghc-9.8`. Easier to just do this.
+source-repository-package
+  type: git
+  location: https://github.com/input-output-hk/aeson
+  tag: 2cb7304e91f17583fc12026762fc5ae6ba9aa1b6
+  subdir:
+    .
+    text-iso8601
+
+source-repository-package
+  type: git
+  -- Git head compiles with `ghc-9.8`, but Hackage version does not.
+  location: https://github.com/recursion-schemes/recursion-schemes
+  tag: cc2e88c3400a6548e975830c9addb12ab087545f
```

Comment:
These need `--sha256` entries. See https://github.com/input-output-hk/ouroboros-consensus/blob/main/CONTRIBUTING.md#use-of-source-repository-packages

Reply:
`aeson` SRP was removed and the other fixes to add a sha256 hash.